subprojects/gradle/buildchecks/src/main/kotlin/com/avito/android/plugin/build_param_check/BuildChecksExtension.kt

```diff
@@ -0,0 +1,98 @@
+package com.avito.android.plugin.build_param_check
+
+import com.avito.android.plugin.build_param_check.BuildChecksExtension.Check.AndroidSdk
+import com.avito.android.plugin.build_param_check.BuildChecksExtension.Check.DynamicDependencies
+import com.avito.android.plugin.build_param_check.BuildChecksExtension.Check.GradleDaemon
+import com.avito.android.plugin.build_param_check.BuildChecksExtension.Check.GradleProperties
+import com.avito.android.plugin.build_param_check.BuildChecksExtension.Check.JavaVersion
+import com.avito.android.plugin.build_param_check.BuildChecksExtension.Check.KotlinApiDependencies
+import com.avito.android.plugin.build_param_check.BuildChecksExtension.Check.ModuleTypes
+import com.avito.android.plugin.build_param_check.BuildChecksExtension.Check.MacOSLocalhost
+import org.gradle.api.Action
+
+open class BuildChecksExtension {
+
+    var enableByDefault: Boolean? = null
+
+    internal val checks = mutableSetOf<Check>()
+
+    fun javaVersion(action: Action<JavaVersion>) = register(JavaVersion(), action)
+
+    fun androidSdk(action: Action<AndroidSdk>) = register(AndroidSdk(), action)
+
+    fun gradleDaemon(action: Action<GradleDaemon>) = register(GradleDaemon(), action)
+
+    fun dynamicDependencies(action: Action<DynamicDependencies>) = register(DynamicDependencies(), action)
+
+    fun macOSLocalhost(action: Action<MacOSLocalhost>) = register(MacOSLocalhost(), action)
+
+    fun gradleProperties(action: Action<GradleProperties>) = register(GradleProperties(), action)
+
+    fun moduleTypes(action: Action<ModuleTypes>) = register(ModuleTypes(), action)
+
+    fun kotlinApiDependencies(action: Action<KotlinApiDependencies>) = register(KotlinApiDependencies(), action)
+
+    private fun <T : Check> register(check: T, action: Action<T>) {
+        @Suppress("UselessCallOnCollection")
+        if (checks.filterIsInstance(check.javaClass).isNotEmpty()) {
+            throw IllegalStateException(
```

Comment:
It is controversial. What is a reason to prevent using twice?
You simply may perform an action on already created check instance.

Reply:
```
buildChecks {
    androidSdk {
        it.compileSdkVersion = 29
        it.revision = 4
    }
}
// ... later in this file
buildChecks {
    androidSdk {
        it.compileSdkVersion = 28
        it.revision = 1
    }
}
```

What should we apply in this case and why? 
In case of overriding we should explain the logic. I prefer to forbid this option until we meet a real need.